File: src/main/java/io/lsdconsulting/lsd/distributed/interceptor/captor/http/derive/PropertyServiceNameDeriver.java
package io.lsdconsulting.lsd.distributed.interceptor.captor.http.derive;

public class PropertyServiceNameDeriver {

    // TODO This should be configurable
    private static final String DEFAULT_SERVICE_NAME_SUFFIX_TO_REMOVE = " Service";

    private final String appName;

    public PropertyServiceNameDeriver(final String appName) {
        this.appName = appName
                .replaceAll(DEFAULT_SERVICE_NAME_SUFFIX_TO_REMOVE, "")
                .replaceAll(" ", "");
    }

    public String getServiceName() {
        return appName;
    }
}

File: src/main/kotlin/io/lsdconsulting/lsd/distributed/interceptor/captor/common/PropertyServiceNameDeriver.kt
package io.lsdconsulting.lsd.distributed.interceptor.captor.common

class PropertyServiceNameDeriver(appName: String) {
    val serviceName: String

    init {
        serviceName = appName
            .replace(DEFAULT_SERVICE_NAME_SUFFIX_TO_REMOVE.toRegex(), "")
            .replace(" ".toRegex(), "")
    }

    companion object {
        // TODO This should be configurable
        private const val DEFAULT_SERVICE_NAME_SUFFIX_TO_REMOVE = " Service"
    }
}
